app/Http/Controllers/Api/ProjectLabelTreeController.php

```diff
@@ -86,19 +86,22 @@ public function index($id)
      * ]
      *
      * @param int $id Project ID
+     * @param string $name Labeltree name
      * @return array<int, LabelTree>
      */
-    public function available($id)
+    public function available($id, $name)
     {
         $project = Project::findOrFail($id);
         $this->authorize('access', $project);
 
         $public = LabelTree::publicTrees()
             ->select('id', 'name', 'description', 'version_id')
+            ->whereRaw('UPPER(name) LIKE UPPER(?)', ["%{$name}%"])
```

Comment:
You could also use:
```php
->where('name', 'ilike', "%{$name}%")
```
This is basically the same but there is less chance to accidentally allow a SQL injection with a raw query. While `ilike` is not supported in all DBMS, BIIGLE exclusively uses Postgres so it is ok.

Reply:
Ah, okay. Could you explain why my approach is still vulnerable to SQL injections? When I was writing the code, this method was recommended as a secure way to prevent SQL injections.

Reply:
It is safe as it is (because you use `?` instead of adding `$name` directly to the string). However, I would only use `whereRaw()` if there is no other way. It's just a matter of risk avoidance.

Reply:
Thanks!